controllers/usersignup/usersignup_controller.go

```diff
@@ -530,19 +530,27 @@ func (r *Reconciler) generateCompliantUsername(config toolchainconfig.ToolchainC
 	for i := 2; i < 101; i++ { // No more than 100 attempts to find a vacant name
 		mur := &toolchainv1alpha1.MasterUserRecord{}
 		// Check if a MasterUserRecord exists with the same transformed name
-		namespacedMurName := types.NamespacedName{Namespace: instance.Namespace, Name: newUsername}
-		err := r.Client.Get(context.TODO(), namespacedMurName, mur)
+		namespacedName := types.NamespacedName{Namespace: instance.Namespace, Name: newUsername}
+		err := r.Client.Get(context.TODO(), namespacedName, mur)
 		if err != nil {
 			if !errors.IsNotFound(err) {
 				return "", err
 			}
-			// If there was a NotFound error looking up the mur, it means we found an available name
-			return newUsername, nil
+			space := &toolchainv1alpha1.Space{}
+			err = r.Client.Get(context.TODO(), namespacedName, space)
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					return "", err
+				}
+				// If there was a NotFound error looking up the mur as well as space, it means we found an available name
+				return newUsername, nil
+			}
```

Comment:
Your code is valid but I believe it would be more readable (and understandable) with something like this:
```suggestion
			if err = r.Client.Get(context.TODO(), namespacedName, space); err != nil {
				if errors.IsNotFound(err) {
					// If there was a NotFound error looking up the mur as well as space, it means we found an available name
					return newUsername, nil
				}
				return "", err
			}
```